pkg/kubelet/dockershim/docker_sandbox_test.go

```diff
@@ -93,64 +93,94 @@ func TestListSandboxes(t *testing.T) {
+			Labels:      test.labels,
+			Annotations: test.annotations,
+		}
 
-	// Check internal labels
-	c, err := fDocker.InspectContainer(id)
-	assert.NoError(t, err)
-	assert.Equal(t, c.Config.Labels[containerTypeLabelKey], containerTypeLabelSandbox)
-	assert.Equal(t, c.Config.Labels[types.KubernetesContainerNameLabel], sandboxContainerName)
+		// Create the sandbox.
+		fClock.SetTime(time.Now())
+		expected.CreatedAt = fClock.Now().UnixNano()
+		runResp, err := ds.RunPodSandbox(getTestCTX(), &runtimeapi.RunPodSandboxRequest{Config: config})
+		require.NoError(t, err)
+		id := runResp.PodSandboxId
 
-	expected.Id = id // ID is only known after the creation.
-	statusResp, err := ds.PodSandboxStatus(getTestCTX(), &runtimeapi.PodSandboxStatusRequest{PodSandboxId: id})
-	require.NoError(t, err)
-	assert.Equal(t, expected, statusResp.Status)
+		// Check internal labels
+		c, err := fDocker.InspectContainer(id)
+		assert.NoError(t, err)
+		assert.Equal(t, c.Config.Labels[containerTypeLabelKey], containerTypeLabelSandbox)
+		assert.Equal(t, c.Config.Labels[types.KubernetesContainerNameLabel], sandboxContainerName)
 
-	// Stop the sandbox.
-	expected.State = runtimeapi.PodSandboxState_SANDBOX_NOTREADY
-	_, err = ds.StopPodSandbox(getTestCTX(), &runtimeapi.StopPodSandboxRequest{PodSandboxId: id})
-	require.NoError(t, err)
-	// IP not valid after sandbox stop
-	expected.Network.Ip = ""
-	statusResp, err = ds.PodSandboxStatus(getTestCTX(), &runtimeapi.PodSandboxStatusRequest{PodSandboxId: id})
-	require.NoError(t, err)
-	assert.Equal(t, expected, statusResp.Status)
+		expected.Id = id // ID is only known after the creation.
+		statusResp, err := ds.PodSandboxStatus(getTestCTX(), &runtimeapi.PodSandboxStatusRequest{PodSandboxId: id})
+		require.NoError(t, err)
+		assert.Equal(t, expected, statusResp.Status)
 
-	// Remove the container.
-	_, err = ds.RemovePodSandbox(getTestCTX(), &runtimeapi.RemovePodSandboxRequest{PodSandboxId: id})
-	require.NoError(t, err)
-	statusResp, err = ds.PodSandboxStatus(getTestCTX(), &runtimeapi.PodSandboxStatusRequest{PodSandboxId: id})
-	assert.Error(t, err, fmt.Sprintf("status of sandbox: %+v", statusResp))
+		// Stop the sandbox.
+		expected.State = runtimeapi.PodSandboxState_SANDBOX_NOTREADY
+		_, err = ds.StopPodSandbox(getTestCTX(), &runtimeapi.StopPodSandboxRequest{PodSandboxId: id})
+		require.NoError(t, err)
+		// IP not valid after sandbox stop
+		expected.Network.Ip = ""
+		statusResp, err = ds.PodSandboxStatus(getTestCTX(), &runtimeapi.PodSandboxStatusRequest{PodSandboxId: id})
+		require.NoError(t, err)
+		assert.Equal(t, expected, statusResp.Status)
+
+		// Remove the container.
+		_, err = ds.RemovePodSandbox(getTestCTX(), &runtimeapi.RemovePodSandboxRequest{PodSandboxId: id})
+		require.NoError(t, err)
+		statusResp, err = ds.PodSandboxStatus(getTestCTX(), &runtimeapi.PodSandboxStatusRequest{PodSandboxId: id})
+		assert.Error(t, err, fmt.Sprintf("status of sandbox: %+v", statusResp))
+	}
```

Comment:
is that too much to test in a single UT? I dont have full context here, but is their a possiblity of breaking this into smaller tests ?